utils/forest_shim/src/error.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+use std::ops::{Deref, DerefMut};
+
+use fvm_shared::error::ExitCode as ExitCodeV2;
+use fvm_shared3::error::ExitCode as ExitCodeV3;
+use serde::{Deserialize, Serialize};
+
+/// Newtype wrapper for FVM `ExitCode`.
+///
+/// # Examples
+/// ```
+/// let fvm2_success = fvm_shared::error::ExitCode::new(0);
+/// let fvm3_success = fvm_shared3::error::ExitCode::new(0);
+///
+/// let shim_from_v2 = forest_shim::error::ExitCode::from(fvm2_success);
+/// let shim_from_v3 = forest_shim::error::ExitCode::from(fvm3_success);
+///
+/// assert_eq!(shim_from_v2, shim_from_v3);
+/// assert_eq!(shim_from_v2, fvm2_success.into());
+/// assert_eq!(*shim_from_v3, fvm3_success);
+/// ```
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct ExitCode(ExitCodeV3);
+
+/// Helper trait to re-use static methods and constants from the `ExitCodeV3`.
+/// The usage is awkward but it avoids code duplication.
+///
+/// ```
+/// use forest_shim::error::{ExitCode, Inner};
+/// <ExitCode as Inner>::FVM::new(0);
+/// ```
+pub trait Inner {
```

Comment:
I think this will be used else (sector info will use it too), maybe move it to `lib.rs`?, and if that's the case maybe the inner associated type can be named something generic.

Reply:
@creativcoder I'll make a separate PR to extract those to a common trait; otherwise, there will be bunch of PRs that should do it. What do you think?

Reply:
Yeah, sounds good

Reply:
@creativcoder I made the change here, I'll adapt other already merged PRs after this one lands in `main`. So don't hesitate to smash this :heavy_check_mark: button. :)